### DISCUSSION TITLE
Should I use "full-height" to force an element to take all remaining space?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#10375] [General - Components / Directives / etc] [answered]
I am writing an app that is made of three components, the second one is a compound one. Please see this playground for details: https://codepen.io/WoJWoJ/pen/NWjJMQb

The status in the playground is that all elements height is fit to their contents. I would like to expand `divB`  to all the available space left on the page. 

I tried various incantations of `full-height`(and others with `grow`, ...) but did not manage to solve this simple problem: having a specific element take all the remaining space.

Specifically, I noted that `full-height` is not relative to the parent, but to the viewport (I believe so because I got very weird effects trying to use several times, at different nesting levels).

**What is the simple, proper way to say "this element should expand to all remaining available space"?**

**Comment by ontwikkelfabriek** [comment, marked as answer]
@wsw70 

Flexbox has your answer, so these links can help:
https://css-tricks.com/snippets/css/a-guide-to-flexbox/
https://flexbox.malven.co/
https://stackoverflow.com/questions/37745051/fill-the-remaining-height-or-width-in-a-flex-container

**Reply by wsw70** [reply]
I already tried that but it does not work (see https://codepen.io/WoJWoJ/pen/JjNzBWK for `flex: 1`added)

**Reply by ontwikkelfabriek** [reply]
@wsw70 

You're **not** using flexbox in your codepen. It's a bit more complicated then adding `flex:1`. You have to set `row`, `col`, `justify-content`, `align-items`, etc. What you want is on a medium complicated level to implement with Flexbox, so not easy peasy.

I would suggest read a bit more about Flexbox in Quasar too:
https://quasar.dev/layout/grid/introduction-to-flexbox

This could help too:
https://quasar.dev/layout/grid/flex-playground

Remember (some of) Quasar flexbox classes/defaults are wrappers/differ from the plain Flexbox classes.

**Reply by wsw70** [reply]
OK, I now realized that Quasar does not set `display: flex` behind the scene *everywhere*. 

I wrote a few toy apps using flexbox and this is why I could not understand why it was not working.

https://codepen.io/WoJWoJ/pen/KKmErwG is an example of a solution to my problem, now that flex is actually applied

**Reply by ontwikkelfabriek** [reply]
@wsw70 

Good to hear you solved it! 

> Quasar does not set display: flex behind the scene everywhere.

Correct  for you own custom components,  but inside Quasar's own visual Components you can assume it is being used almost always. Can be useful information when you are dealing with **slots** defined inside Quasar's components.
